server/routes/link/VerifyLinkController.ts

```diff
@@ -26,4 +28,26 @@ export default class VerifyLinkController {
         res.redirect('request-link')
       })
   }
+
+  private verifyToken(token: string, req: Request, res: Response) {
+    verify(token, config.barcodeTokenPublicKey, { algorithms: ['RS256'] }, (err: VerifyErrors, payload: JwtPayload) => {
+      if (err) {
+        req.flash('errors', [
+          { href: '#email', text: 'The link you used is no longer valid. Request a new one to sign in.' },
+        ])
+        res.redirect('/link/request-link')
+      }
+      req.session.validSlmToken = true
+      req.session.barcodeUserEmail = payload.sub
+      req.session.barcodeUserOrganisation = payload.organisation
+      // make the session expiry the same as the JWT - otherwise we lose the JWT when the session expires
+      req.session.cookie.expires = new Date(payload.exp * 1000)
+      const { id, ...sessionWithoutId } = req.session
+      req.session.regenerate(() => {
+        // put the old session back - but keep the new session ID
+        Object.assign(req.session, sessionWithoutId)
+        res.redirect('/barcode/find-recipient')
+      })
```

Comment:
This fixes the session fixation vulnerability... which is the thing I really wanted to test but as the fix is inside a callback I'm not quite sure how 😢